app/src/main/java/org/wikipedia/page/edithistory/EditHistoryListViewModel.kt

```diff
@@ -92,14 +92,20 @@ class EditHistoryListViewModel(savedStateHandle: SavedStateHandle) : ViewModel()
             val editCountsUserResponse = async { ServiceFactory.getCoreRest(pageTitle.wikiSite).getEditCount(pageTitle.prefixedText, EditCount.EDIT_TYPE_EDITORS) }
             val editCountsAnonResponse = async { ServiceFactory.getCoreRest(pageTitle.wikiSite).getEditCount(pageTitle.prefixedText, EditCount.EDIT_TYPE_ANONYMOUS) }
             val editCountsBotResponse = async { ServiceFactory.getCoreRest(pageTitle.wikiSite).getEditCount(pageTitle.prefixedText, EditCount.EDIT_TYPE_BOT) }
-            val articleMetricsResponse = async { ServiceFactory.getRest(WikiSite("wikimedia.org")).getArticleMetrics(pageTitle.wikiSite.authority(), pageTitle.prefixedText, lastYear, today) }
+            val articleMetricsResponse = async {
+                try {
+                    ServiceFactory.getRest(WikiSite("wikimedia.org")).getArticleMetrics(pageTitle.wikiSite.authority(), pageTitle.prefixedText, lastYear, today)
+                } catch (_: Exception) {
```

Comment:
I feel like this is too blunt of a hammer for too small of a nut.
* It's generally not recommended to catch the base `Exception` anyway, and it's _really_ not recommended to catch `Exception` inside a coroutine, because then we risk catching `CancellationException` which is thrown when a coroutine is cancelled.
* If this API only works for Main-namespace articles, then can we just check that `pageTitle.namespace() == Namespace.MAIN)` here?

Reply:
Thanks! It makes sense to me. Done.